tasks/DroneFlightControlTask.cpp

```diff
@@ -76,38 +141,30 @@ void DroneFlightControlTask::updateHook()
 
     auto control_device =
         mVehicle->subscribe->getValue<Telemetry::TOPIC_CONTROL_DEVICE>();
+
+    States new_state = runtimeStatesTransition(control_device);
+    applyTransition(new_state);
```

Comment:
Doesn't this mean you're calling `obtainCtrlAuthority` every single `updateHook` ?